rewrite-maven/src/main/java/org/openrewrite/maven/internal/MavenPomDownloader.java

```diff
@@ -139,6 +140,9 @@ private MavenPomDownloader(Map<Path, Pom> projectPoms, HttpSender httpSender, Ex
         this.projectPomsByGav = projectPomsByGav(projectPoms);
         this.httpSender = httpSender;
         this.ctx = MavenExecutionContextView.view(ctx);
+        if (ctx.getMessage(MAVEN_SETTINGS_LOAD_FROM_DISK, false)) {
```

Comment:
I now made it work with this flag. You could argue though, if there is no maven-settings available in the context, you always want to read the maven settings from local. What do you guys think?

Reply:
Let's do that. It's a sensible default behavior and then we don't need to toggle this boolean just right.